schema/deploy/application_validation_functions/no_reported_ciip_product.sql

```diff
@@ -0,0 +1,34 @@
+-- Deploy ggircs-portal:application_validation_functions/no_reported_ciip_product to pg
+-- requires: tables/application_revision_validation_function
+-- requires: computed_columns/application_revision_production_form_data
+-- requires: computed_columns/application_revision_validation
+
+begin;
+
+create or replace function ggircs_portal.no_reported_ciip_product(app_revision ggircs_portal.application_revision)
+  returns boolean
+  as $$
+
+    select exists (
+      select * from ggircs_portal.application_revision_production_form_data(app_revision) pfd
+      join ggircs_portal.product p
+      on pfd.product_id = p.id
+      and p.is_ciip_product = true
+      and p.is_energy_product = false
+      and pfd.product_amount > 0
+    );
+
+$$ language sql stable;
+
+grant execute on function ggircs_portal.no_reported_ciip_product to ciip_administrator, ciip_analyst, ciip_industry_user;
+
+comment on function ggircs_portal.no_reported_ciip_product(ggircs_portal.application_revision) is 'This validation function for a CIIP (CleanBC Industrial Incentive Program) application determines if it contains at least one valid ciip product reported';
+
+insert into ggircs_portal.application_revision_validation_function(validation_function_name, validation_description, validation_failed_message)
```

Comment:
Note that, as we discovered with @kriscooke while looking at #1644 , items that are inserted in the migrations will not be present in e2e tests, as the `test_helper.clean_ggircs_portal_schema` function will empty all the tables.
An sql file in `schema/data/prod` inserting those might be preferable, especially if we want to update those descriptions/failed messages later

Reply:
Does this mean I should hold off on the proposed fix in my test fixture?

Reply:
I think so. Adding an script in `schema/data/prod` might be a better solution for cases where some table records are required for features that we are developing (advanced validation functions, application review steps, form schemas). Inserts in the sqitch migrations seem to be a better option for seed data (e.g. the list of naics codes) that can be modified later by a user of the application. 

I'll add a follow up to my comment in #1644